/**
*					--DAO-Generator--
*
* Java code generated by the DAO-Generator developed by akcess(www.akcess.in)
* Date of code generation: Tue Jun 22 00:02:11 ART 2010
*
* DAO-Generator Version 1.2
*
*/
package metalsoft.datos.idao;
import java.math.*;
import java.sql.*;
import java.net.URL;
import java.util.*;
import metalsoft.datos.exception.*;
import metalsoft.datos.dbobject.*;
public interface ResponsableDAO
{
	public int insert(Responsable responsable, Connection con ) throws ResponsableException;
	public int update(ResponsablePK responsablepk, Responsable responsable, Connection con) throws ResponsableException;
	public int delete(ResponsablePK responsablepk, Connection con) throws ResponsableException;
	public Responsable findByPrimaryKey(ResponsablePK responsablepk , Connection con) throws ResponsableException;
	public Responsable[] findAll(Connection con) throws ResponsableException;
	public Responsable[] findByIdresponsable(long idresponsable,Connection con) throws ResponsableException;
	public Responsable[] findByNombre(String nombre,Connection con) throws ResponsableException;
	public Responsable[] findByApellido(String apellido,Connection con) throws ResponsableException;
	public Responsable[] findByTelefono(String telefono,Connection con) throws ResponsableException;
	public Responsable[] findByEmail(String email,Connection con) throws ResponsableException;
	public Responsable[] findByDomicilio(long domicilio,Connection con) throws ResponsableException;
	public Responsable[] findByNrodocumento(int nrodocumento,Connection con) throws ResponsableException;
	public Responsable[] findByTipodocumento(long tipodocumento,Connection con) throws ResponsableException;
	public Responsable[] findByFax(String fax,Connection con) throws ResponsableException;
	public Responsable[] findExecutingUserSelect(String selectStatement, Object[] sqlParams, Connection con) throws ResponsableException;
	public Responsable[] findExecutingUserWhere(String whereClause, Object[] sqlParams, Connection con) throws ResponsableException;
}
